scripts/benchmarks_ci.py

```diff
@@ -221,25 +222,37 @@ def __main(args: list) -> int:
 
     # Run micro-benchmarks
     if not args.build_only:
-        for framework in args.frameworks:
-            micro_benchmarks.run(
-                BENCHMARKS_CSPROJ,
-                args.configuration,
-                framework,
-                verbose,
-                args
-            )
-            
-        dotnet.shutdown_server(verbose)
-
-        if args.upload_to_perflab_container:
-            import upload
+        upload_container = UPLOAD_CONTAINER
+        try:
+            for framework in args.frameworks:
+                micro_benchmarks.run(
+                    BENCHMARKS_CSPROJ,
+                    args.configuration,
+                    framework,
+                    verbose,
+                    args
+                )
             globpath = os.path.join(
                 get_artifacts_directory() if not args.bdn_artifacts else args.bdn_artifacts,
                 '**',
                 '*perf-lab-report.json')
+        except CalledProcessError:
+            upload_container = 'failedresults'
+            globpath = os.path.join(
+                get_artifacts_directory() if not args.bdn_artifacts else args.bdn_artifacts,
+                'FailureReporter', 
+                'failure-report.json')
+            cmdline = [
+                'FailureReporting.exe', globpath
+            ]
+            RunCommand(cmdline, verbose=verbose).run(
```

Comment:
After we create the path to the executable, we should check to make sure it exists, and if it doesn't we should throw here so we don't upload something that we do not intend. This should also happen in the startup.py file.